crates/web5/src/dids/methods/web/resolver.rs

```diff
@@ -0,0 +1,116 @@
+use std::{
+    future::{Future, IntoFuture},
+    pin::Pin,
+};
+
+use reqwest::header::{HeaderMap, HeaderValue};
+
+use crate::dids::{
+    document::Document,
+    identifier::Identifier,
+    resolver::{ResolutionError, ResolutionResult},
+};
+
+// PORT_SEP is the : character that separates the domain from the port in a URI.
+const PORT_SEP: &str = "%3A";
+
+const USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"));
```

Comment:
Mostly because it's a way for the did:web server to understand what versions/libraries it's dealing with, though given it's not a service-to-service environment and there are many callers maybe it's not as important. I'm not strong on keeping it or removing it -- I only added it because it was a natural thing for me to include, but also happy to not have it included.

Re. WASM environments -- that's a good questions. I'll implement it in a browser/WASM environment soon for the DWN project I'm working on, and if this is included I can report back (otherwise, I'll probably have other places where I make `reqwest` requests in the DWN so I can try it there). One challenge with WASM in general is networking, but WASIX and other modern preview specs include some networking (including HTTP stubbed by `fetch`, iirc)

Reply:
I'm wrangling around code into [the APID](https://github.com/TBD54566975/web5-rs/pull/229) in [this PR](https://github.com/TBD54566975/web5-rs/pull/234). I'm going to go ahead and merge this code and copy everything over into the temporary `apid` module, but I'm going to remove this feature. It's probably something we want, but I want to approach these matters conservatively because of downstream impacts to cross-language binding. I appreciate the thought! And I suspect we'll add it back at some point once we fully comprehend the implications.